src/SnapshotTesting.ts

```diff
@@ -0,0 +1,88 @@
+import { Input } from './Input'
+import * as lsif from './lsif'
+import { Range } from './Range'
+
+const lsiftyped = lsif.lib.codeintel.lsiftyped
+
+export function formatSnapshot(
+  input: Input,
+  document: lsif.lib.codeintel.lsiftyped.Document
+): string {
+  const out: string[] = []
+  document.occurrences.sort(occurrencesByLine)
+  const symbolTable = new Map<
+    string,
+    lsif.lib.codeintel.lsiftyped.SymbolInformation
+  >()
+  for (const symbolInfo of document.symbols) {
+    symbolTable.set(symbolInfo.symbol, symbolInfo)
+  }
+  let occurrenceIndex = 0
+  for (const [lineNumber, line] of input.lines.entries()) {
+    out.push('  ')
+    out.push(line)
+    out.push('\n')
+    while (
+      occurrenceIndex < document.occurrences.length &&
+      document.occurrences[occurrenceIndex].range[0] === lineNumber
+    ) {
+      const occurrence = document.occurrences[occurrenceIndex]
+      occurrenceIndex++
+      if (occurrence.range.length > 3) {
+        // Skip multiline occurrences for now.
+        continue
+      }
+      const range = Range.fromLsif(occurrence.range)
+      out.push('//')
+      out.push(' '.repeat(range.start.character))
+      const length = range.end.character - range.start.character
+      if (length < 0) {
+        throw new Error(input.format(range, 'negative length occurrence!'))
+      }
+      out.push('^'.repeat(length))
+      out.push(' ')
+      const isDefinition =
+        (occurrence.symbol_roles & lsiftyped.SymbolRole.Definition) > 0
```

Comment:
Maybe `!!` or `!== 0` would be more idiomatic here rather than `> 0`?

Reply:
I'm not sure what's most idiomatic, this is how I would have written it in Java :/